### ISSUE TITLE
`swarm update` with an invalid docker image version gives "Wrong Input"

### ISSUE THREAD

**Issue by ewoutp** [giantswarm/cli#378]
With `swarm update` you are supposed to only pass a new version. If you pass a full image name instead, it will given "Wrong Input." instead of a proper error message.

E.g.
```
$ swarm update redis-example-expose/example/db/redis busybox:ubuntu-14.04
Updating component 'redis-example-expose/example/db/redis' with version 'busybox:ubuntu-14.04' and strategy 'one-by-one'...
Wrong Input.
```

**Comment by marians** [comment]
I suggest to ignore everything before the colon, so `busybox:ubuntu-14.04` is parsed as `ubuntu-14.04`, to prevent this case from happening.

**Comment by ewoutp** [comment]
I assume we should only do that if the image name is the same as what is in the definition.
If so, I like the idea.

**Comment by marians** [comment]
Good point.

Here is a message for the other case (image names unequal):

```
Bad image name '<image_name_given_by_user>' provided.
The component uses the image '<current_image_name>' and can only be updated to versions (tags) of this image.
```

`image_name_given_by_user` and `current_image_name` should not contain the version tag, only the part before the colon.

**Comment by marians** [comment]
Closed via #525
